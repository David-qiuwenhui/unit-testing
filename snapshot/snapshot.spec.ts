import { it, expect, describe } from "vitest";
import Hi from "./Hi.vue";
import app from "./app.vue";
import { mount } from "@vue/test-utils";

describe("snapshot", () => {
  it("snapshot object ", () => {
    expect({
      a: 1,
      b: 2,
      c: 3,
    }).toMatchSnapshot();
  });

  it("snapshot array", () => {
    expect([1, 2, 3]).toMatchSnapshot();
  });

  it.todo("snapshot object ", () => {
    expect({
      a: 1,
      b: 2,
      c: 3,
    }).toMatchInlineSnapshot(`
      {
        "a": 1,
        "b": 2,
        "c": 3,
      }
    `);
  });

  it("snapshot component", () => {
    const wrapper = mount(Hi);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("snapshot component file", () => {
    const wrapper = mount(app);
    expect(wrapper.html()).toMatchFileSnapshot("./app.html");
  });
});
