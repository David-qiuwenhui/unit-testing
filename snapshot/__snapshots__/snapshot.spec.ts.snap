// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot > snapshot array 1`] = `
[
  1,
  2,
  3,
]
`;

exports[`snapshot > snapshot component 1`] = `"<div id="hi">hi how are you</div>"`;

exports[`snapshot > snapshot object  1`] = `
{
  "a": 1,
  "b": 2,
  "c": 3,
}
`;
